reaktive/src/nativeCommonMain/kotlin/com/badoo/reaktive/single/SingleByEmitter.kt

```diff
@@ -0,0 +1,47 @@
+package com.badoo.reaktive.single
+
+import com.badoo.reaktive.base.tryCatch
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+
+// Separate implementation prevents unnecessary freezing: https://github.com/badoo/Reaktive/issues/472
+actual inline fun <T> single(crossinline onSubscribe: (emitter: SingleEmitter<T>) -> Unit): Single<T> =
+    singleUnsafe { observer ->
+        val disposableWrapper = DisposableWrapper()
+        observer.onSubscribe(disposableWrapper)
```

Comment:
The only difference, that you extracted `DisposableWrapper` from `SingleEmitter`?
Can't we have this implementation for all platforms then?

Reply:
Yes, the JVM version allocates less objects and so more optimal. I made this optimization when I was fighting with benchmarks.

Reply:
@CherryPerry Whay do you think?